src/jit/codegencommon.cpp

```diff
@@ -6343,13 +6329,27 @@ void            CodeGen::genPopCalleeSavedRegistersAndFreeLclFrame(bool
         if ((compiler->lvaOutgoingArgSpaceSize == 0) && (totalFrameSize < 512))
         {
             frameType = 1;
+            if (compiler->compLocallocUsed)
+            {
+                // Restore sp from fp
+                //      mov sp, fp
+                inst_RV_RV(INS_mov, REG_SPBASE, REG_FPBASE);
+            }
 
```

Comment:
Every single instruction in the epilog needs an unwind code. You don't have any here, or below, in your new code.